internal/handlers/pg/pgdb/query_iterator.go

```diff
@@ -98,6 +98,9 @@ func (it *queryIterator) Close() {
 		return
 	}
 
-	it.rows.Close()
+	if it.rows != nil {
```

Comment:
fwiw, it could be wrapped with sync.Once instead of mutex and flag

Reply:
Ah, that's the function I tried to remember but couldn't! I'll make changes in this PR then.

Reply:
fixed